cmake/modules/hunter_calculate_config_sha1.cmake

```diff
@@ -128,6 +128,11 @@ function(hunter_calculate_config_sha1 hunter_self hunter_base user_config)
         )
       endif()
 
+      string(COMPARE NOTEQUAL "{HUNTER_${x}_KEEP_PACKAGE_SOURCES}" "" have_keep_package_sources)
```

Comment:
I don't see the point of changing `config-id`. Logically it's not affected.

Reply:
I see.  The `set` call in `hunter_config. cmake` was being cl

When I had the following log at the bottom of `hunter_config.cmake`
```
  message("MOD: ${_hunter_current_project}  ${HUNTER_${_hunter_current_project}_KEEP_PACKAGE_SOURCES}")
```

It was clearing getting cleared on the third call.  

`grep MOD _logs/polly/default/log.txt  | grep GTest`

```
MOD: GTest
MOD: GTest  ON
MOD: GTest
```

So the package source wasn't being kept.  I noticed the third call to `hunter_config()` was coming from this generated file, and this mod fixed it:

```
MOD: GTest
MOD: GTest  ON
MOD: GTest  ON  # fixed
``` 

This all seems to be due to the misplaced variable in hunter_config.cmake [here](https://github.com/ruslo/hunter/pull/1216/files/13fdc60b17e81560b0c5e7cc0e24ebf53927ae36#r151868052).